sklearn/base.py

```diff
@@ -115,6 +116,10 @@ def _clone_parametrized(estimator, *, safe=True):
 
     params_set = new_object.get_params(deep=False)
 
+    # attach callbacks to the new estimator
+    if hasattr(estimator, "_callbacks"):
```

Comment:
> BTW, this might break something for https://github.com/microsoft/FLAML/blob/0415638dd1e1d3149fb17fb8760520af975d16f6/flaml/automl/model.py#L1587 which also adds this attribute to scikit-learn's base estimator in their library

I can change the name to _sk_callbacks or any derivative of that.

Reply:
> Why aren't we creating a copy here?

The motivation is a use case like this
```py
monitoring = Monitoring()
lr = LogisitcRegression()._set_callbacks(monitoring)
GridSearchCV(lr, param_grid).fit(X,y)
monitoring.plot()
```
The monitoring callback will gather information across all copies of logistic regression made in the grid search. If we made a copy of the callback in `clone`, then we couldn't retrieve any information once the grid search is finished.

Reply:
The object itself can disable copy by implementing `__copy__` and `__deep_copy__`, and then they would be in the space of "we know what we're doing" and we don't need to worry about it.

I think the kind of state you're referring to here, is something which can be outside the callback object, like a file / a database / an external singleton object, and the callback method just writes into that storage, and at the end one can use that data to plot/investigate/etc.

Reply:
> I think the kind of state you're referring to here, is something which can be outside the callback object, like a file / a database / an external singleton object

good point and actually in my latest design of the callbacks I didn't rely on a shared state so you can ignore my previous comment :) And we can't get around copies anyway since the clones can happen in subprocesses (in a grid search for instance). I updated to clone the callbacks as any other param